ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/StandaloneExecutorTest.java

```diff
@@ -40,54 +55,125 @@
-
-    executor.start();
 
+    final QueryMetadata csas1QueryMetadata = EasyMock.niceMock(PersistentQueryMetadata.class);
+    EasyMock.expect(csas1QueryMetadata.getDataSourceType()).andReturn(DataSourceType.KSTREAM);
+    EasyMock.expect(engine.buildMultipleQueries(eq("CSAS1"), anyObject(KsqlConfig.class), anyObject(Map.class)))
+        .andReturn(Collections.singletonList(csas1QueryMetadata)).once();
+    EasyMock.expect(engine.getQueryExecutionPlan(csas1Query, ksqlConfig))
+        .andReturn(csas1QueryMetadata).once();
+
+    final QueryMetadata ctasQueryMetadata = EasyMock.niceMock(PersistentQueryMetadata.class);
+    EasyMock.expect(ctasQueryMetadata.getDataSourceType()).andReturn(DataSourceType.KTABLE);
+    EasyMock.expect(engine.buildMultipleQueries(eq("CTAS"), anyObject(KsqlConfig.class), anyObject(Map.class)))
+        .andReturn(Collections.singletonList(ctasQueryMetadata)).once();
+    EasyMock.expect(engine.getQueryExecutionPlan(ctasQuery, ksqlConfig))
+        .andReturn(ctasQueryMetadata).once();
+
+    final QueryMetadata csas2QueryMetadata = EasyMock.niceMock(PersistentQueryMetadata.class);
+    EasyMock.expect(csas2QueryMetadata.getDataSourceType()).andReturn(DataSourceType.KSTREAM);
+    EasyMock.expect(engine.buildMultipleQueries(eq("CSAS2"), anyObject(KsqlConfig.class), anyObject(Map.class)))
+        .andReturn(Collections.singletonList(csas2QueryMetadata)).once();
+    EasyMock.expect(engine.getQueryExecutionPlan(csas2Query, ksqlConfig))
+        .andReturn(csas1QueryMetadata).once();
+
+    EasyMock.replay(setProperty);
+    EasyMock.replay(csas1Query);
+    EasyMock.replay(createStreamAsSelect1);
+    EasyMock.replay(csas1QueryMetadata);
+    EasyMock.replay(ctasQuery);
+    EasyMock.replay(createTableAsSelect);
+    EasyMock.replay(ctasQueryMetadata);
+    EasyMock.replay(csas2Query);
+    EasyMock.replay(createStreamAsSelect2);
+    EasyMock.replay(csas2QueryMetadata);
+    EasyMock.replay(engine);
+    standaloneExecutor.start();
     EasyMock.verify(engine);
   }
 
-  @Test
-  public void shouldExecutePersistentQueries() {
-    final PersistentQueryMetadata query = EasyMock.niceMock(PersistentQueryMetadata.class);
-    EasyMock.expect(engine.createQueries(anyString(), eq(ksqlConfig)))
-        .andReturn(Collections.singletonList(query));
-    query.start();
-    EasyMock.expectLastCall();
-    EasyMock.replay(query, engine);
-
-    executor.start();
-
-    EasyMock.verify(query);
-  }
+  @Test(expected = KsqlException.class)
+  public void shouldFailIfCsasIsInvalid() {
+    final StandaloneExecutor standaloneExecutor =
+        new StandaloneExecutor(ksqlConfig, engine, queriesFile, udfLoader);
+
+    final List<Pair<String, Statement>> statementPairs = new ArrayList<>();
+    final MetaStore metaStore = EasyMock.niceMock(MetaStore.class);
+    statementPairs.add(new Pair<>("CS", EasyMock.niceMock(CreateStream.class)));
+    statementPairs.add(new Pair<>("CT", EasyMock.niceMock(CreateTable.class)));
```

Comment:
Why do we need this line? Aren't we just testing if the `CSAS` is invalid?

Reply:
Removed them.